modules.cmake

```diff
@@ -43,25 +43,83 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif ()
 
 
+
+# Receives latest available C++ standard version
+#
+# Usage:
+#   modules_get_latest_cxx(<variable_name>)
+#   if(<variable_name> GREATER 17)
+#      ...
+#   endif()
+function(modules_get_latest_cxx result)
+  foreach(ver 26 23 20 17 14 11 98)
+    if(CMAKE_CXX_COMPILE_FEATURES MATCHES "cxx_std_${ver}")
+      set(${result} ${ver} PARENT_SCOPE)
+      break()
+    endif()
+  endforeach()
+endfunction()
+
+
+# Checks that compiler has support for C++ modules feature
+#
+# Usage:
+#   modules_is_supported(<variable_name>)
+#   if(<variable_name>)
+#      ...
+#   endif()
+function(modules_is_supported result)
+    set(${result} FALSE PARENT_SCOPE)
+
+    # check standard version
+    modules_get_latest_cxx(latest_standard)
+    if(latest_standard GREATER_EQUAL 20)
+
+      # create simple module file
+      set(temp_filepath "${CMAKE_CURRENT_BINARY_DIR}/module_test.cc")
```

Comment:
Where does CMake usually generate files like this?

Reply:
Since CMake 3.25: `try_compile(SOURCE_FROM_VAR)`
Before CMake 3.25: `${CMAKE_BINARY_DIR}`

So, location changed and added `file(REMOVE ...)` to clean up stuff.